girder/utility/acl_mixin.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright 2015 Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+import itertools
+import six
+
+from ..models.model_base import Model
+from ..constants import AccessType
+
+class AccessControlMixin(object):
+    """
+    This mixin is intended to be used for resources which aren't access
+    controlled by default, but resolve their access controls through other
+    resources.
+
+    resource_coll corresponds to the resource collection that needs to be used
+    for resolution, for example an item would resolve its resource_coll as
+    folder (if it weren't already access controlled).
+
+    resource_parent corresponds to the field in which the parent resource
+    belongs, so for an item it would be the folderId.
+    """
+    resource_coll = None
+    resource_parent = None
```

Comment:
For consistency, let's use camelcase for these. I know one of the style PEPs discourages that, but all of our other variable names are that way so we are kind of stuck now :)